src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -92,56 +92,57 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
     }
 
-    private bool ShouldDropMessage(CmsEvent cmsEvent)
+    private bool IsValidComponent(ContentComponentDbEntity mapped)
     {
-        if (cmsEvent == CmsEvent.CREATE)
-        {
-            Logger.LogInformation("Dropping received event {cmsEvent}", cmsEvent);
-            return true;
-        }
-
-        if (ShouldIgnoreMessage(cmsEvent))
-        {
-            Logger.LogInformation("Receieved {event} but UsePreview is {usePreview} - dropping message", cmsEvent, _contentfulOptions.UsePreview);
-            return true;
-        }
-
-        return false;
-    }
-
-    private bool IsNewAndValidComponent(ContentComponentDbEntity mapped, ContentComponentDbEntity? existing)
-    {
-        if (existing != null) return true;
-
-        if (AnyRequiredPropertyIsNull(mapped))
+        if (AnyRequiredPropertyIsNull(mapped, out List<PropertyInfo?> nullProperties))
         {
-            Logger.LogInformation("A required property of the component with ID {id} is null, so the message will be dropped!", mapped.Id);
+            Logger.LogInformation("Content Component with ID {id} is missing the following required properties: {nullProperties}", mapped.Id, string.Join(", ", nullProperties));
             return false;
         }
 
         return true;
     }
 
-    private bool AnyRequiredPropertyIsNull(ContentComponentDbEntity entity)
-    => _db.Model.FindEntityType(entity.GetType())!
-        .GetProperties()
-        .Where(prop => !prop.IsNullable)
-        .Select(prop => prop.PropertyInfo)
-        .Where(prop => !prop!.CustomAttributes.Any(atr => atr.GetType() == typeof(DontCopyValueAttribute)))
-        .Where(prop => prop!.GetValue(entity) == null)
-        .Any();
+    private bool AnyRequiredPropertyIsNull(ContentComponentDbEntity entity, out List<PropertyInfo?> nullProperties)
```

Comment:
I'd put this back to `.Any()`

With what you've got:
- It'll iterate through every item in the `IEnumerable`
- It'll add it to a list

With `Any()` it'll iterate through the `IEnumerable` until it finds any item matching the predicate, then it'll return true. If there's no item matching it'll return false.

With the former you're guaranteed you'll always get n iterations, and you'll be allocating a list unnecessarily (since you're only getting the count from it).

I assume you've done this because Sonarcloud (or something else) has recommended it because I've seen it before. It's wrong!

Reply:
Okay I see the intention now, I missed that sorry!

In which case IMO something like this would be better:
<img width="1112" alt="image" src="https://github.com/DFE-Digital/plan-technology-for-your-school/assets/131891854/de541e5f-9d0c-4e60-ad34-87d9459e93b2">

It'll prevent the list being allocated, which could potentially be quite large (as it'll be created with a capacity matching the item count being iterated on).

It'll also allow flexibility in its usage.
- Instead of building a string of the null properties, we could just call `.Any()` if needed.
- If we wanted to get null properties and then do something else on them it'd allow us to do that (e.g. if we wanted to filter the properties further, say by ones that are strings only, we could call it then add `.Where(property => property.PropertyType == typeof(string))` etc). 
- We could map the propertyinfos to something else with `.Select`
- Only take X count
- etc.

Reply:
Resolved in: [858d341](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/508/commits/858d341d6cc83ec33e7e7979743a80ceb4a5c03c)